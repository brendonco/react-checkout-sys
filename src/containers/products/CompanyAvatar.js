import React from 'react';

const CompanyAvatar = ({
  match:{
    params: { companyId = "default" } = {}
  }
}) => {
  return <img className="company__avatar" src={require(`assets/images/${companyId}-logo.svg`)} alt='avatar' />;
};

export default CompanyAvatar;
