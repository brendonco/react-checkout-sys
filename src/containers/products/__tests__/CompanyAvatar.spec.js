import React from 'react';
import renderer from 'react-test-renderer';
import CompanyAvatar from 'containers/products/CompanyAvatar';

describe('Render different avatar', () => {
  it('Render list of avatar', () => {

    [{
      match:{
        params: { companyId: "unilever" }
      }
    },
    {
      match:{
        params: { companyId: "apple" }
      }
    },{
      match:{
        params: { companyId: "nike" }
      }
    },{
      match:{
        params: { companyId: "ford" }
      }
    }, {
      match:{
        params: {}
      }
    }].forEach(el =>{
      const component = renderer.create(<CompanyAvatar {...el} />); 
      expect(component).toMatchSnapshot();
    });
  });
})