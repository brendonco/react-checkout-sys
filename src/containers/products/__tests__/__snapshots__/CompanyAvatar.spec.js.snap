// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render different avatar Render list of avatar 1`] = `
<img
  alt="avatar"
  className="company__avatar"
  src="unilever-logo.svg"
/>
`;

exports[`Render different avatar Render list of avatar 2`] = `
<img
  alt="avatar"
  className="company__avatar"
  src="apple-logo.svg"
/>
`;

exports[`Render different avatar Render list of avatar 3`] = `
<img
  alt="avatar"
  className="company__avatar"
  src="nike-logo.svg"
/>
`;

exports[`Render different avatar Render list of avatar 4`] = `
<img
  alt="avatar"
  className="company__avatar"
  src="ford-logo.svg"
/>
`;

exports[`Render different avatar Render list of avatar 5`] = `
<img
  alt="avatar"
  className="company__avatar"
  src="default-logo.svg"
/>
`;
